smack-experimental/src/main/java/org/jivesoftware/smackx/mam/element/MamV2ElementFactory.java

```diff
@@ -0,0 +1,69 @@
+/**
+ *
+ * Copyright © 2016-2021 Florian Schmaus and Frank Matheron
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.mam.element;
+
+import java.util.List;
+import javax.xml.namespace.QName;
+
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smackx.forward.packet.Forwarded;
+import org.jivesoftware.smackx.rsm.packet.RSMSet;
+import org.jivesoftware.smackx.xdata.packet.DataForm;
+
+import org.jxmpp.jid.Jid;
+
+public class MamV2ElementFactory implements MamElementFactory {
+
+    MamV2ElementFactory() {
```

Comment:
As with `MamV1ElementFactory`'s constructor.

Reply:
I changed the class visibility to package private and removed the (empty) constructor.